Rubberduck.Core/UI/Command/CommandBase.cs

```diff
@@ -53,10 +53,20 @@ public bool CanExecute(object parameter)
             }
         }
 
+        protected virtual bool RequireReEvaluationOnExecute => false;
+
         public void Execute(object parameter)
         {
             try
             {
+                if (RequireReEvaluationOnExecute)
+                {
+                    if (!CanExecute(parameter))
```

Comment:
I like that idea - in reality we only need to test the `Terminated` and not necessarily other conditions because it is possible to have that changed between the check and the execution, and we'd then get disconnection errors anyway. 

I will refactor the functions to have the optional bool parameter.